### PR TITLE
DOCSUP-9674: Document the intervalLengthSum aggregate function

### PR DESCRIPTION
**Pull request by sevirov**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

Changelog category:

- Documentation for #24250

### CODE REVIEW

File: docs/en/sql-reference/aggregate-functions/reference/intervalLengthSum.md [hunk 1]
```diff
@@ -0,0 +1,143 @@
+---
+toc_priority: 146
+toc_title: intervalLengthSum
+---
+
+# intervalLengthSum {#agg_function-intervallengthsum}
+
+Calculates the sum of the length of all ranges (segments on numeric axis) without counting intersection twice.
```
Comment (adevyatova) [inline docs/en/sql-reference/aggregate-functions/reference/intervalLengthSum.md:8]:
```suggestion
Calculates the sum of the length of all ranges (segments on numeric axis) excluding intersections.
```

File: docs/en/sql-reference/aggregate-functions/reference/intervalLengthSum.md [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+
+**Syntax**
+
+``` sql
+intervalLengthSum(start, end)
+```
+
+**Arguments**
+
+-   `start` — Starting interval value. [Int32](../../../sql-reference/data-types/int-uint.md#uint8-uint16-uint32-uint64-int8-int16-int32-int64), [Int64](../../../sql-reference/data-types/int-uint.md#uint8-uint16-uint32-uint64-int8-int16-int32-int64), [UInt32](../../../sql-reference/data-types/int-uint.md#uint8-uint16-uint32-uint64-int8-int16-int32-int64), [UInt64](../../../sql-reference/data-types/int-uint.md#uint8-uint16-uint32-uint64-int8-int16-int32-int64), [Float32](../../../sql-reference/data-types/float.md#float32-float64), [Float64](../../../sql-reference/data-types/float.md#float32-float64), [DateTime](../../../sql-reference/data-types/datetime.md#data_type-datetime) or [Date](../../../sql-reference/data-types/date.md#data_type-date).
```
Comment (adevyatova) [inline docs/en/sql-reference/aggregate-functions/reference/intervalLengthSum.md:18]:
```suggestion
-   `start` — The starting value of the interval. [Int32](../../../sql-reference/data-types/int-uint.md#uint8-uint16-uint32-uint64-int8-int16-int32-int64), [Int64](../../../sql-reference/data-types/int-uint.md#uint8-uint16-uint32-uint64-int8-int16-int32-int64), [UInt32](../../../sql-reference/data-types/int-uint.md#uint8-uint16-uint32-uint64-int8-int16-int32-int64), [UInt64](../../../sql-reference/data-types/int-uint.md#uint8-uint16-uint32-uint64-int8-int16-int32-int64), [Float32](../../../sql-reference/data-types/float.md#float32-float64), [Float64](../../../sql-reference/data-types/float.md#float32-float64), [DateTime](../../../sql-reference/data-types/datetime.md#data_type-datetime) or [Date](../../../sql-reference/data-types/date.md#data_type-date).
```

File: docs/en/sql-reference/aggregate-functions/reference/intervalLengthSum.md [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+-   `end` — Ending interval value. [Int32](../../../sql-reference/data-types/int-uint.md#uint8-uint16-uint32-uint64-int8-int16-int32-int64), [Int64](../../../sql-reference/data-types/int-uint.md#uint8-uint16-uint32-uint64-int8-int16-int32-int64), [UInt32](../../../sql-reference/data-types/int-uint.md#uint8-uint16-uint32-uint64-int8-int16-int32-int64), [UInt64](../../../sql-reference/data-types/int-uint.md#uint8-uint16-uint32-uint64-int8-int16-int32-int64), [Float32](../../../sql-reference/data-types/float.md#float32-float64), [Float64](../../../sql-reference/data-types/float.md#float32-float64), [DateTime](../../../sql-reference/data-types/datetime.md#data_type-datetime) or [Date](../../../sql-reference/data-types/date.md#data_type-date).
```
Comment (adevyatova) [inline docs/en/sql-reference/aggregate-functions/reference/intervalLengthSum.md:19]:
```suggestion
-   `end` — The ending value of the interval. [Int32](../../../sql-reference/data-types/int-uint.md#uint8-uint16-uint32-uint64-int8-int16-int32-int64), [Int64](../../../sql-reference/data-types/int-uint.md#uint8-uint16-uint32-uint64-int8-int16-int32-int64), [UInt32](../../../sql-reference/data-types/int-uint.md#uint8-uint16-uint32-uint64-int8-int16-int32-int64), [UInt64](../../../sql-reference/data-types/int-uint.md#uint8-uint16-uint32-uint64-int8-int16-int32-int64), [Float32](../../../sql-reference/data-types/float.md#float32-float64), [Float64](../../../sql-reference/data-types/float.md#float32-float64), [DateTime](../../../sql-reference/data-types/datetime.md#data_type-datetime) or [Date](../../../sql-reference/data-types/date.md#data_type-date).
```

File: docs/en/sql-reference/aggregate-functions/reference/intervalLengthSum.md [hunk 4]
```diff
@@ -0,0 +1,143 @@
...
+
+!!! info "Note"
+    Arguments must be of the same data type. Otherwise, an exception will be thrown.
+
+**Returned value**
+
+-   Sum of the length of all ranges (segments on numeric axis) without counting intersection twice. If the arguments are of integer type, the function returns a value of the [UInt64](../../../sql-reference/data-types/int-uint.md#uint8-uint16-uint32-uint64-int8-int16-int32-int64) type. If the arguments are of the floating-point type, the function returns a value of the [Float64](../../../sql-reference/data-types/float.md#float32-float64) type.
```
Comment (adevyatova) [inline docs/en/sql-reference/aggregate-functions/reference/intervalLengthSum.md:26]:
```suggestion
-   Sum of the length of all ranges (segments on numeric axis) excluding intersections. Depending on the type of the argument, the return value may be [UInt64](../../../sql-reference/data-types/int-uint.md#uint8-uint16-uint32-uint64-int8-int16-int32-int64) or [Float64](../../../sql-reference/data-types/float.md#float32-float64) type.
```

File: docs/en/sql-reference/aggregate-functions/reference/intervalLengthSum.md [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+1. Input table:
+
+``` text
+┌─id─┬─start─┬─end─┐
+│ a  │   1.1 │ 3.2 │
+│ a  │     4 │   5 │
+└────┴───────┴─────┘
+```
+
+In this example, the arguments of the `Float32` type are used. In this case, the function returns a value of the `Float64` type:
```
Comment (adevyatova) [inline docs/en/sql-reference/aggregate-functions/reference/intervalLengthSum.md:39]:
```suggestion
In this example, the arguments of the Float32 type are used. The function returns a value of the Float64 type.

Query:
```

File: docs/en/sql-reference/aggregate-functions/reference/intervalLengthSum.md [hunk 6]
```diff
@@ -0,0 +1,97 @@
...
+2. Input table:
+
+``` text
+┌─id─┬───────────────start─┬─────────────────end─┐
+│ a  │ 2020-01-01 01:12:30 │ 2020-01-01 02:50:31 │
+│ a  │ 2020-01-01 03:11:22 │ 2020-01-01 03:23:31 │
+└────┴─────────────────────┴─────────────────────┘
+```
+
+In this example, the arguments of the `DateTime` type are used. In this case, the function returns a value in seconds:
```
Comment (adevyatova) [inline docs/en/sql-reference/aggregate-functions/reference/intervalLengthSum.md:62]:
```suggestion
In this example, the arguments of the DateTime type are used. The function returns a value in seconds.

Query:
```

File: docs/en/sql-reference/aggregate-functions/reference/intervalLengthSum.md [hunk 7]
```diff
@@ -0,0 +1,97 @@
...
+3. Input table:
+
+``` text
+┌─id─┬──────start─┬────────end─┐
+│ a  │ 2020-01-01 │ 2020-01-04 │
+│ a  │ 2020-01-12 │ 2020-01-18 │
+└────┴────────────┴────────────┘
+```
+
+In this example, the arguments of the `Date` type are used. In this case, the function returns a value in days:
```
Comment (adevyatova) [inline docs/en/sql-reference/aggregate-functions/reference/intervalLengthSum.md:85]:
```suggestion
In this example, the arguments of the Date type are used. The function returns a value in days.

Query:
```

**Comment by sevirov** [comment]
@vdimir Please, review this PR.

File: docs/en/sql-reference/aggregate-functions/reference/intervalLengthSum.md [hunk 8]
```diff
@@ -0,0 +1,103 @@
...
+-   Sum of the length of all ranges (segments on numeric axis) excluding intersections. Depending on the type of the argument, the return value may be [UInt64](../../../sql-reference/data-types/int-uint.md#uint8-uint16-uint32-uint64-int8-int16-int32-int64) or [Float64](../../../sql-reference/data-types/float.md#float32-float64) type.
+
+**Examples**
+
+1. Input table:
+
+``` text
+┌─id─┬─start─┬─end─┐
+│ a  │   1.1 │ 3.2 │
+│ a  │     4 │   5 │
```
Comment (vdimir) [inline docs/en/sql-reference/aggregate-functions/reference/intervalLengthSum.md:35]: Example with intersections more interesting
```suggestion
│ a  │   1.1 │ 2.9 │
│ a  │   2.5 │ 3.2 │
│ a  │     4 │   5 │
```

File: docs/ru/sql-reference/aggregate-functions/reference/intervalLengthSum.md [hunk 9]
```diff
@@ -0,0 +1,103 @@
...
+-   Сумма длин всех интервалов (отрезков на числовой оси), исключая пересечения. В зависимости от типа аргумента возвращаемое значение может быть типа [UInt64](../../../sql-reference/data-types/int-uint.md#uint8-uint16-uint32-uint64-int8-int16-int32-int64) или [Float64](../../../sql-reference/data-types/float.md#float32-float64).
+
+**Примеры**
+
+1. Входная таблица:
+
+``` text
+┌─id─┬─start─┬─end─┐
+│ a  │   1.1 │ 3.2 │
+│ a  │     4 │   5 │
```
Comment (vdimir) [inline docs/ru/sql-reference/aggregate-functions/reference/intervalLengthSum.md:35]:
```suggestion
│ a  │   1.1 │ 2.9 │
│ a  │   2.5 │ 3.2 │
│ a  │     4 │   5 │
```

File: docs/ru/sql-reference/aggregate-functions/reference/intervalLengthSum.md [hunk 10]
```diff
@@ -0,0 +1,103 @@
...
+│ a  │                          3.1 │ Float64                                  │
+└────┴──────────────────────────────┴──────────────────────────────────────────┘
+```
+
+2. Входная таблица:
+
+``` text
+┌─id─┬───────────────start─┬─────────────────end─┐
+│ a  │ 2020-01-01 01:12:30 │ 2020-01-01 02:50:31 │
+│ a  │ 2020-01-01 03:11:22 │ 2020-01-01 03:23:31 │
```
Comment (vdimir) [inline docs/ru/sql-reference/aggregate-functions/reference/intervalLengthSum.md:60]:
```suggestion
│ a  │ 2020-01-01 01:12:30 │ 2020-01-01 02:10:10 │
│ a  │ 2020-01-01 02:05:30 │ 2020-01-01 02:50:31 │
│ a  │ 2020-01-01 03:11:22 │ 2020-01-01 03:23:31 │
```

File: docs/en/sql-reference/aggregate-functions/reference/intervalLengthSum.md [hunk 11]
```diff
@@ -0,0 +1,103 @@
...
+│ a  │                          3.1 │ Float64                                  │
+└────┴──────────────────────────────┴──────────────────────────────────────────┘
+```
+
+2. Input table:
+
+``` text
+┌─id─┬───────────────start─┬─────────────────end─┐
+│ a  │ 2020-01-01 01:12:30 │ 2020-01-01 02:50:31 │
+│ a  │ 2020-01-01 03:11:22 │ 2020-01-01 03:23:31 │
```
Comment (vdimir) [inline docs/en/sql-reference/aggregate-functions/reference/intervalLengthSum.md:60]:
```suggestion
│ a  │ 2020-01-01 01:12:30 │ 2020-01-01 02:10:10 │
│ a  │ 2020-01-01 02:05:30 │ 2020-01-01 02:50:31 │
│ a  │ 2020-01-01 03:11:22 │ 2020-01-01 03:23:31 │
```

File: docs/en/sql-reference/aggregate-functions/reference/intervalLengthSum.md [hunk 12]
```diff
@@ -0,0 +1,103 @@
...
+1. Input table:
+
+``` text
+┌─id─┬─start─┬─end─┐
+│ a  │   1.1 │ 3.2 │
+│ a  │     4 │   5 │
+└────┴───────┴─────┘
+```
+
+In this example, the arguments of the Float32 type are used. The function returns a value of the Float64 type.
```
Comment (vdimir) [inline docs/en/sql-reference/aggregate-functions/reference/intervalLengthSum.md:39]:
```suggestion
In this example, the arguments of the Float32 type are used. The function returns a value of the Float64 type.

Result is the sum of lengths of intervals `[1.1, 3.2]` (union of `[1.1, 2.9]` and `[2.5, 3.2]`) and `[4, 5]`
```

File: docs/ru/sql-reference/aggregate-functions/reference/intervalLengthSum.md [hunk 13]
```diff
@@ -0,0 +1,103 @@
...
+1. Входная таблица:
+
+``` text
+┌─id─┬─start─┬─end─┐
+│ a  │   1.1 │ 3.2 │
+│ a  │     4 │   5 │
+└────┴───────┴─────┘
+```
+
+В этом примере используются аргументы типа Float32. Функция возвращает значение типа Float64.
```
Comment (vdimir) [inline docs/ru/sql-reference/aggregate-functions/reference/intervalLengthSum.md:39]:
```suggestion
В этом примере используются аргументы типа Float32. Функция возвращает значение типа Float64.

Результатом функции будет сумма длин интервалов `[1.1, 3.2]` (объединение `[1.1, 2.9]` и `[2.5, 3.2]`) и `[4, 5]`
```

File: docs/en/sql-reference/aggregate-functions/reference/intervalLengthSum.md [hunk 14]
```diff
@@ -0,0 +1,103 @@
+---
+toc_priority: 146
+toc_title: intervalLengthSum
+---
+
+# intervalLengthSum {#agg_function-intervallengthsum}
+
+Calculates the sum of the length of all ranges (segments on numeric axis) excluding intersections.
```
Comment (vdimir) [inline docs/en/sql-reference/aggregate-functions/reference/intervalLengthSum.md:8]: 'excluding intersections' sounds like we do not count intersections at all, but we count it once
```suggestion
Calculates the total length of union of all ranges (segments on numeric axis).
```

File: docs/ru/sql-reference/aggregate-functions/reference/intervalLengthSum.md [hunk 15]
```diff
@@ -0,0 +1,103 @@
+---
+toc_priority: 146
+toc_title: intervalLengthSum
+---
+
+# intervalLengthSum {#agg_function-intervallengthsum}
+
+Вычисляет сумму длин всех интервалов (отрезков на числовой оси), исключая пересечения.
```
Comment (vdimir) [inline docs/ru/sql-reference/aggregate-functions/reference/intervalLengthSum.md:8]:
```suggestion
Вычисляет длину объединения множества интервалов (отрезков на числовой оси).
```

File: docs/ru/sql-reference/aggregate-functions/reference/intervalLengthSum.md [hunk 16]
```diff
@@ -0,0 +1,103 @@
...
+
+!!! info "Примечание"
+    Аргументы должны быть одного типа. В противном случае ClickHouse сгенерирует исключение.
+
+**Возвращаемое значение**
+
+-   Сумма длин всех интервалов (отрезков на числовой оси), исключая пересечения. В зависимости от типа аргумента возвращаемое значение может быть типа [UInt64](../../../sql-reference/data-types/int-uint.md#uint8-uint16-uint32-uint64-int8-int16-int32-int64) или [Float64](../../../sql-reference/data-types/float.md#float32-float64).
```
Comment (vdimir) [inline docs/ru/sql-reference/aggregate-functions/reference/intervalLengthSum.md:26]:
```suggestion
-   Длина объединения всех интервалов (отрезков на числовой оси). В зависимости от типа аргумента возвращаемое значение может быть типа [UInt64](../../../sql-reference/data-types/int-uint.md#uint8-uint16-uint32-uint64-int8-int16-int32-int64) или [Float64](../../../sql-reference/data-types/float.md#float32-float64).
```

File: docs/en/sql-reference/aggregate-functions/reference/intervalLengthSum.md [hunk 17]
```diff
@@ -0,0 +1,103 @@
...
+
+!!! info "Note"
+    Arguments must be of the same data type. Otherwise, an exception will be thrown.
+
+**Returned value**
+
+-   Sum of the length of all ranges (segments on numeric axis) excluding intersections. Depending on the type of the argument, the return value may be [UInt64](../../../sql-reference/data-types/int-uint.md#uint8-uint16-uint32-uint64-int8-int16-int32-int64) or [Float64](../../../sql-reference/data-types/float.md#float32-float64) type.
```
Comment (vdimir) [inline docs/en/sql-reference/aggregate-functions/reference/intervalLengthSum.md:26]:
```suggestion
-   Total length of union of all ranges (segments on numeric axis). Depending on the type of the argument, the return value may be [UInt64](../../../sql-reference/data-types/int-uint.md#uint8-uint16-uint32-uint64-int8-int16-int32-int64) or [Float64](../../../sql-reference/data-types/float.md#float32-float64) type.
```

File: docs/en/sql-reference/aggregate-functions/reference/intervalLengthSum.md [hunk 18]
```diff
@@ -0,0 +1,103 @@
...
+``` sql
+SELECT id, intervalLengthSum(start, end), toTypeName(intervalLengthSum(start, end)) FROM fl_interval GROUP BY id ORDER BY id;
+```
+
+Result:
+
+``` text
+┌─id─┬─segmentLengthSum(start, end)─┬─toTypeName(segmentLengthSum(start, end))─┐
+│ a  │                          3.1 │ Float64                                  │
+└────┴──────────────────────────────┴──────────────────────────────────────────┘
```
Comment (vdimir) [inline docs/en/sql-reference/aggregate-functions/reference/intervalLengthSum.md:52]:
```suggestion
┌─id─┬─intervalLengthSum(start, end)─┬─toTypeName(intervalLengthSum(start, end))─┐
│ a  │                           3.1 │ Float64                                   │
└────┴───────────────────────────────┴───────────────────────────────────────────┘
```

File: docs/ru/sql-reference/aggregate-functions/reference/intervalLengthSum.md [hunk 19]
```diff
@@ -0,0 +1,103 @@
...
+``` sql
+SELECT id, intervalLengthSum(start, end), toTypeName(intervalLengthSum(start, end)) FROM fl_interval GROUP BY id ORDER BY id;
+```
+
+Результат:
+
+``` text
+┌─id─┬─segmentLengthSum(start, end)─┬─toTypeName(segmentLengthSum(start, end))─┐
+│ a  │                          3.1 │ Float64                                  │
+└────┴──────────────────────────────┴──────────────────────────────────────────┘
```
Comment (vdimir) [inline docs/ru/sql-reference/aggregate-functions/reference/intervalLengthSum.md:52]:
```suggestion
┌─id─┬─intervalLengthSum(start, end)─┬─toTypeName(intervalLengthSum(start, end))─┐
│ a  │                           3.1 │ Float64                                   │
└────┴───────────────────────────────┴───────────────────────────────────────────┘
```

**Review by vdimir — APPROVED** [review]

**Pull request merged**